react/components/EXPERIMENTAL_TableTree/index.tsx

```diff
@@ -1,38 +1,60 @@
 import React, { FC } from 'react'
+import PropTypes from 'prop-types'
 
-import { TableProvider } from '../EXPERIMENTAL_Table/context'
+import { TableContext } from '../EXPERIMENTAL_Table/contexts'
+import { CheckboxesContext } from './contexts'
 import Toolbar from '../EXPERIMENTAL_Table/Toolbar'
 import DataTable from '../EXPERIMENTAL_Table/DataTable'
 import Tree from './Tree'
 import { TableContainer, Thead } from '../EXPERIMENTAL_Table/Styled'
 import { tablePropTypes, TableComposites } from '../EXPERIMENTAL_Table'
 import { InferProps } from 'prop-types'
+import TreeHeadings from './Tree/TreeHeadings'
 
 const TableTree: FC<Props> & TableComposites = ({
   children,
   state,
+  checkboxes,
   ...props
 }) => {
   return (
-    <TableProvider value={{ ...state, ...props }}>
-      <TableContainer>
-        {children}
-        <DataTable>
-          <Thead>
-            <DataTable.Headings />
-          </Thead>
-          <tbody>
-            <Tree />
-          </tbody>
-        </DataTable>
-      </TableContainer>
-    </TableProvider>
+    <TableContext.Provider value={{ ...state, ...props }}>
+      <CheckboxesContext.Provider value={{ ...checkboxes }}>
+        <TableContainer>
+          {children}
+          <DataTable>
+            <Thead>
+              <TreeHeadings />
+            </Thead>
+            <tbody>
+              <Tree />
+            </tbody>
+          </DataTable>
+        </TableContainer>
+      </CheckboxesContext.Provider>
+    </TableContext.Provider>
   )
 }
 
-type Props = InferProps<typeof tablePropTypes>
+const itemTreeShape = PropTypes.shape({
+  children: PropTypes.any,
```

Comment:
```suggestion
  children: PropTypes.node,
``` ?

Reply:
Nope, this is children of an item. Node doesn't fit

Reply:
children shouldn't be anything that is rederizable?

Reply:
I share your concern since this can cause confusion with the `children` prop of react components. That's why now I added a prop to allow the user to choose the key that will contain each item children.